x/hard/keeper/repay.go

```diff
@@ -17,9 +20,6 @@ func (k Keeper) Repay(ctx sdk.Context, sender, owner sdk.AccAddress, coins sdk.C
 	// Call incentive hook
 	k.BeforeBorrowModified(ctx, borrow)
```

Comment:
We intentionally changed the ordering in https://github.com/Kava-Labs/kava/pull/800#discussion_r571646085 and https://github.com/Kava-Labs/kava/pull/804 so that `BeforeBorrowModified` is called before `SyncBorrowInterest`. Bit of a chicken and egg problem here. Does modifying the borrow before calling the `BeforeBorrowModified` incentive hook cause any issues? We'll likely want to follow a similar pattern in borrow.go, repay.go, deposit.go, and withdraw.go.

Reply:
This came up in CDP accumulator audit - it's correct to call the hook before interest is updated.